packages/store/ts/config/v2/store.test.ts

```diff
@@ -563,46 +595,53 @@ describe("resolveStoreConfig", () => {
         ValidNames: ["first", "second"],
       },
     });
-    attest<{
+    const expected = {
       tables: {
         Example: {
           schema: {
             key: {
-              type: "string";
-              internalType: "dynamic";
-            };
+              type: "string",
+              internalType: "dynamic",
+            },
             name: {
-              type: "uint8";
-              internalType: "ValidNames";
-            };
+              type: "uint8",
+              internalType: "ValidNames",
+            },
             age: {
-              type: "address";
-              internalType: "static";
-            };
-          };
+              type: "address",
+              internalType: "static",
+            },
+          },
           keySchema: {
             name: {
-              type: "uint8";
-              internalType: "ValidNames";
-            };
-          };
+              type: "uint8",
+              internalType: "ValidNames",
+            },
+          },
           valueSchema: {
             age: {
-              type: "address";
-              internalType: "static";
-            };
+              type: "address",
+              internalType: "static",
+            },
             key: {
-              type: "string";
-              internalType: "dynamic";
-            };
-          };
-          primaryKey: ["name"];
-        };
-      };
-      userTypes: { static: "address"; dynamic: "string" };
+              type: "string",
+              internalType: "dynamic",
+            },
+          },
+          primaryKey: ["name"],
+        },
+      },
+      userTypes: { static: "address", dynamic: "string" },
       enums: {
-        ValidNames: ["first", "second"];
-      };
-    }>(config);
+        ValidNames: ["first", "second"],
+      },
+      namespace: "",
+    } as const;
+    attest<typeof expected>(config);
+  });
+
+  it("should use the root namespace as default namespace", () => {
+    const config = resolveStoreConfig({ tables: { Example: {} } });
+    attest<"">(config.namespace);
```

Comment:
should we have a test for when namespace is set?

Reply:
also I thought namespaces were meant to be a world thing, not a store thing?

Reply:
protocol wise yes, but tablegen also needs namespaces, so it feels fine to add it to the store config (so it can be part of the expected resolved table config)